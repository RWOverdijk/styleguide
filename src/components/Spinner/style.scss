@import '../../css/variables';
@import '../../css/mixins';

$animation-duration: 4s;

.Van-Spinner {
  background-color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  left: 0;
  padding: 25px;
  text-align: center;
  width: 100%;

  &--fullScreen {
    height: 100vh;
    position: fixed;
    top: 0;
    width: 100vw;
    z-index: 9; // the react-header has z-index: 10
  }

  &-iconContainer {
    height: 50px;
  }

  &-icon {
    animation: zoom-in-out $animation-duration ease-out infinite;
    left: 0;
    position: absolute;
    right: 0;
    transform: scale(0);
    transform-origin: 50% 100%;
    width: 100%;

    > svg {
      height: 250%;
    }
  }

  &-sun {
    @include color-fill($color-yellow-primary);
  }

  &-wind {
    @include color-fill($color-pink-primary);

    animation-delay: $animation-duration * 0.25;
  }

  &-hydro {
    @include color-fill($color-blue-primary);

    animation-delay: $animation-duration / 2;
  }

  &-bio {
    @include color-fill($color-green-primary);

    animation-delay: $animation-duration * 0.75;
  }
}

@keyframes zoom-in-out {
  0% {
    transform: scale(0);
  }

  15% {
    transform: scale(1);
  }

  18% {
    transform: scale(1);
  }

  30% {
    transform: scale(0);
  }
}
