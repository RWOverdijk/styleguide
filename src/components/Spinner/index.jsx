import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Icon from '../Icon';
import iconBio from '../../assets/icons/ftic/bio.svg';
import iconHydro from '../../assets/icons/ftic/hydro.svg';
import iconSun from '../../assets/icons/ftic/sun.svg';
import iconWind from '../../assets/icons/ftic/wind.svg';
import './style.scss';

const Spinner = ({ children, fullScreen }) => (
  <div className={classnames('Van-Spinner', { 'Van-Spinner--FullScreen': fullScreen })}>
    <div className="Van-Spinner-IconContainer">
      <Icon source={iconBio} className="Van-Spinner-Icon Van-Spinner-Bio" />
      <Icon source={iconHydro} className="Van-Spinner-Icon Van-Spinner-Hydro" />
      <Icon source={iconSun} className="Van-Spinner-Icon Van-Spinner-Sun" />
      <Icon source={iconWind} className="Van-Spinner-Icon Van-Spinner-Wind" />
    </div>

    {children && <div className="Van-Spinner-Content">{children}</div>}
  </div>
);

Spinner.defaultProps = {
  children: null,
  fullScreen: false
};

Spinner.propTypes = {
  children: PropTypes.node,
  fullScreen: PropTypes.bool
};

export default Spinner;
