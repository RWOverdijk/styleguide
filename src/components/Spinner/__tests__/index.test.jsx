import React from 'react';
import renderer from 'react-test-renderer';
import Spinner from '../';

describe('<Spinner />', () => {
  it('renders a spinner', () => {
    expect(renderer.create(
      <Spinner />
    ).toJSON()).toMatchSnapshot();
  });

  it('renders a spinner with children', () => {
    expect(renderer.create(
      <Spinner>
        <strong>i am a child</strong>
      </Spinner>
    ).toJSON()).toMatchSnapshot();
  });

  it('renders a full screen spinner', () => {
    expect(renderer.create(
      <Spinner fullScreen />
    ).toJSON()).toMatchSnapshot();
  });
});
