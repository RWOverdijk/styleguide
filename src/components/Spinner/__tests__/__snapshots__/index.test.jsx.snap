// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> renders a full screen spinner 1`] = `
<div
  className="Van-Spinner Van-Spinner--FullScreen"
>
  <div
    className="Van-Spinner-IconContainer"
  />
</div>
`;

exports[`<Spinner /> renders a spinner 1`] = `
<div
  className="Van-Spinner"
>
  <div
    className="Van-Spinner-IconContainer"
  />
</div>
`;

exports[`<Spinner /> renders a spinner with children 1`] = `
<div
  className="Van-Spinner"
>
  <div
    className="Van-Spinner-IconContainer"
  />
  <div
    className="Van-Spinner-Content"
  >
    <strong>
      i am a child
    </strong>
  </div>
</div>
`;
